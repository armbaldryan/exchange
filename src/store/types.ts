export interface ICurrency {
	[key: string]: string;
}

export interface ITimeSeries {
	[key: string]: {
		[key: string]: {
			[key: string]: {
				[key: string]: number;
			};
		};
	};
}

export enum ActionTypesEnum {
	SET_ALL_CURRENCY,
	ADD_TIMESERIES,
}

export interface StateModel {
	allCurrency?: ICurrency;
	timeseries: ITimeSeries;
}

export interface ActionModel {
	type: ActionTypesEnum;
	payload: any;
}

export enum DateTimeEnum {
	year,
	month,
	halfYear,
}
