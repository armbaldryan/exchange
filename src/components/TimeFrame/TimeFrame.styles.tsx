import * as React from "react";

import { styled } from "@mui/material/styles";
import Button, { ButtonProps } from "@mui/material/Button";

export const TimeFrameWrapper = styled("div")(
	() => `
		display: flex;
		flexDirection: row;
		margin: 30px 0;
	`
);

export const TimeButton = styled((props: ButtonProps) => <Button {...props} />)(
	() => `
  		margin-right: 10px;
		border-radius: 0;
	`
);
