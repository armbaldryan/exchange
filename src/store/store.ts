import { createContext, useContext } from "react";

import { StateModel, ActionTypesEnum, ActionModel } from "./types";

export const initialState: StateModel = {
	timeseries: {},
};

export const reducer = (
	state: StateModel,
	{ type, payload }: ActionModel
): StateModel => {
	switch (type) {
		case ActionTypesEnum.SET_ALL_CURRENCY:
			return { ...state, allCurrency: payload };
		case ActionTypesEnum.ADD_TIMESERIES:
			return {
				...state,
				timeseries: {
					...state.timeseries,
					[payload.key]: {
						...(state.timeseries[payload.key] || {}),
						[payload.dateTime]: payload.rates,
					},
				},
			};
		default:
			return state;
	}
};

export const Store = createContext({
	state: initialState,
	dispatch: (args: any) => {},
});

export const useStore = () => useContext(Store);
