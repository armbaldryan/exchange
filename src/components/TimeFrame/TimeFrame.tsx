import React, { MouseEventHandler } from "react";

import { TimeButton, TimeFrameWrapper } from "./TimeFrame.styles";
import { DateTimeEnum } from "store/types";

type Props = {
	dateTime?: number;
	changeTimeFrame: MouseEventHandler<HTMLButtonElement>;
};

const TimeFrame = (props: Props) => {
	const { changeTimeFrame, dateTime } = props;

	return (
		<TimeFrameWrapper data-testid="time-frame-wrapper">
			<TimeButton
				data-testid="year"
				name={DateTimeEnum.year.toString()}
				onClick={changeTimeFrame}
				variant={dateTime === DateTimeEnum.year ? "contained" : "outlined"}
			>
				Year
			</TimeButton>
			<TimeButton
				data-testid="half-year"
				name={DateTimeEnum.halfYear.toString()}
				onClick={changeTimeFrame}
				variant={dateTime === DateTimeEnum.halfYear ? "contained" : "outlined"}
			>
				6 Months
			</TimeButton>
			<TimeButton
				data-testid="month"
				name={DateTimeEnum.month.toString()}
				onClick={changeTimeFrame}
				variant={dateTime === DateTimeEnum.month ? "contained" : "outlined"}
			>
				Month
			</TimeButton>
		</TimeFrameWrapper>
	);
};

export default TimeFrame;
