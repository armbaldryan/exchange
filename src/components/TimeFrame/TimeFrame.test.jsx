import TimeFrame from "./TimeFrame";
import React from "react";
import { render } from "@testing-library/react";
import { DateTimeEnum } from "store/types";

describe("Time Frame", () => {
	it("should render Time Frame", () => {
		const { container } = render(
			<TimeFrame dateTime={DateTimeEnum.year} changeTimeFrame={jest.fn()} />
		);
		expect(container).toMatchSnapshot();
	});

	it("should select Year button", () => {
		const { container, getByTestId } = render(
			<TimeFrame dateTime={DateTimeEnum.year} changeTimeFrame={jest.fn()} />
		);
		const wrapper = getByTestId("time-frame-wrapper");
		expect(wrapper.firstChild).toHaveClass("MuiButton-containedSizeMedium");
	});

	it("should select half year button", () => {
		const { container, getByTestId } = render(
			<TimeFrame dateTime={DateTimeEnum.halfYear} changeTimeFrame={jest.fn()} />
		);
		const wrapper = getByTestId("half-year");
		expect(wrapper).toHaveClass("MuiButton-containedSizeMedium");
	});

	it("should select month button", () => {
		const { container, getByTestId } = render(
			<TimeFrame dateTime={DateTimeEnum.month} changeTimeFrame={jest.fn()} />
		);
		const wrapper = getByTestId("month");
		expect(wrapper).toHaveClass("MuiButton-containedSizeMedium");
	});
});
