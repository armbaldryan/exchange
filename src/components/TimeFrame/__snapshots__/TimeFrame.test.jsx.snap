// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time Frame should render Time Frame 1`] = `
<div>
  <div
    class="css-j7miut"
    data-testid="time-frame-wrapper"
  >
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-qiyftw-MuiButtonBase-root-MuiButton-root"
      data-testid="year"
      name="0"
      tabindex="0"
      type="button"
    >
      Year
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-14aompz-MuiButtonBase-root-MuiButton-root"
      data-testid="half-year"
      name="2"
      tabindex="0"
      type="button"
    >
      6 Months
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-14aompz-MuiButtonBase-root-MuiButton-root"
      data-testid="month"
      name="1"
      tabindex="0"
      type="button"
    >
      Month
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
